packages/data/README.md

```diff
@@ -133,8 +120,12 @@ A **resolver** is a side-effect for a selector. If your selector result may need
 
 The `resolvers` option should be passed as an object where each key is the name of the selector to act upon, the value a function which receives the same arguments passed to the selector, excluding the state argument. It can then dispatch as necessary to fulfill the requirements of the selector, taking advantage of the fact that most data consumers will subscribe to subsequent state changes (by `subscribe` or `withSelect`).
 
+Resolvers, in combination with [thunks](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-can-be-async), can be used to implement asynchronous data flows for your store. 
+
 #### `controls`
 
+> To handle asynchronous data flows, it is recommended to use [thunks](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/thunks.md#thunks-can-be-async) instead of `controls` now that they are supported.
+
 A **control** defines the execution flow behavior associated with a specific action type. This can be particularly useful in implementing asynchronous data flows for your store. By defining your action creator or resolvers as a generator which yields specific controlled action types, the execution will proceed as defined by the control handler.
 
 The `controls` option should be passed as an object where each key is the name of the action type to act upon, the value a function which receives the original action object. It should returns either a promise which is to resolve when evaluation of the action should continue, or a value. The value or resolved promise value is assigned on the return value of the yield assignment. If the control handler returns undefined, the execution is not continued.
```

Comment:
I am not sure what to do with the `controls` section. Should it be refactored? Removed? I am not familiar with how controls are used.

Reply:
Can we hide it in a "details" kind of? and mark it as "deprecated"?

Reply:
I've tried that in this commit: https://github.com/WordPress/gutenberg/pull/67006/commits/6f24f40701503a5e2e16dd5f017130a841139de4?short_path=f874ff4#diff-f874ff4a9c60876a3d01ac27950807424eead31c2cd914daa34136108bd03e0f. Let me know if that's not what you had in mind. I am not sure if it needs to be changed somewhere else.

Reply:
I saw that the Interactivity API readmes have a bunch of `<details>` ([example](https://github.com/WordPress/gutenberg/blob/6f24f40701503a5e2e16dd5f017130a841139de4/docs/reference-guides/interactivity-api/api-reference.md#wp-context)) and they seem to look fine in the block editor handbook ([example](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/api-reference/#wp-context)). Although it's true in this case I'm adding different elements.

Reply:
It makes perfect sense to mark the controls as deprecated 👍🏻